src/nomad_simulations/general.py

```diff
@@ -177,6 +179,31 @@ def _set_system_branch_depth(
             system_child.branch_depth = branch_depth + 1
             self._set_system_branch_depth(system_child, branch_depth + 1)
 
+    def resolve_composition_formula(
+        self, system_parent: ModelSystem, logger: BoundLogger
+    ) -> None:
+        """
```

Comment:
There is a very nice extension in VSCode (autoDocstring) that, together with Github copilot, essentially generates the docstring of these functions for you. Maybe you want to check this out